File: 6.rb
def format_ordinal(number)
  hundred_units = number % 100
  if hundred_units == 11 || hundred_units == 12 || hundred_units == 13
    return "#{number}th"
  end

  digit = number % 10
  case digit
  when 1
    "#{number}st"
  when 2
    "#{number}nd"
  when 3
    "#{number}rd"
  else
    "#{number}th"
  end
end

(1..25).each do |n|
  puts format_ordinal(n)
end

puts format_ordinal(100)
puts format_ordinal(101)
puts format_ordinal(102)
puts format_ordinal(103)
puts format_ordinal(104)

puts format_ordinal(111)
puts format_ordinal(112)
puts format_ordinal(113)

puts format_ordinal(213)

File: 6.js
function formatOrdinal(number) {
  const hundredUnits = number % 100
  if (hundredUnits === 11 || hundredUnits === 12 || hundredUnits === 13) {
    return `${number}th`
  }

  const digit = number % 10
  switch (digit) {
    case 1:
      return `${number}st`
    case 2:
      return `${number}nd`
    case 3:
      return `${number}rd`
    default:
      return `${number}th`
  }
}

for (let n = 1; n <= 25; n++) {
  console.log(formatOrdinal(n))
}

console.log(formatOrdinal(100))
console.log(formatOrdinal(101))
console.log(formatOrdinal(102))
console.log(formatOrdinal(103))
console.log(formatOrdinal(104))

console.log(formatOrdinal(111))
console.log(formatOrdinal(112))
console.log(formatOrdinal(113))

console.log(formatOrdinal(213))
